tests/test_sync.py

```diff
@@ -268,6 +280,17 @@ def test_sync_board_jira_to_zen(self, zenhub_post, zenhub_put, github_patch):
                                                                      'body': 'synchronized with github: Repository Name: abc Issue Number: 4',
                                                                      'labels': []})
 
+    @patch('src.sync.Sync.sync_from_specified_source')
+    def test_mirror_sync(self, sync):
+        """Assert that two issues are synced from Jira to ZenHub and two from ZenHub to Jira, based on timestamps."""
+        Sync.mirror_sync(self.JIRA_REPO, self.ZENHUB_REPO)
+
+        # Call args list has the addresses of issues being synced, which change each time, so just look at issue type
+        called_with = [(call[0][0].__class__.__name__, call[0][1].__class__.__name__) for call in sync.call_args_list]
+        expected = [('JiraIssue', 'ZenHubIssue'), ('JiraIssue', 'ZenHubIssue'), ('ZenHubIssue', 'JiraIssue'),
+                    ('ZenHubIssue', 'JiraIssue')]
+        self.assertEqual(called_with, expected)
+
     def tearDown(self):
         patch.stopall()  # Stop all the patches that were started in setUp
```

Comment:
This is a bit tedious but for the sake of curiosity: The [docs say](https://docs.python.org/3.3/library/unittest.mock.html#patch-methods-start-and-stop) that `patch.stopall()` only stops patches that were started with `patch.start()`. Wondering whether patch used as a decorator is similar to a context manager (and thus doesn't need to be explicitly stopped)?

Reply:
You're right, this line only affects the patches started in setUp(). In setUp() I used patch.start() instead of the decorators so that they would apply to the whole test suite, also pycharm doesn't like having multiple arguments in the setUp definition. Not sure this is the best way